Automattic/crowdsignal-ui
# Add/email form block

This PR adds a block variation to the text form block.

A new file was created that holds an array of variation objects.  Adding additional variations to the text input form can be done by adding new objects to the text form block.

Another new file, validation.js was created to conveniently hold variation code and corresponds with the validation attribute in the variations file.

A function call was added to the block/text-input/index.js file that will run the function named in the variations object to run any sort of form field validation related to the new type of field.

This last commit fixes the last of the suggestions in the previous PR, and this is ready for another look.

Testing instructions:

Checkout branch and yarn install/start
Add a new project via https://crowdsignal.localhost:9000/project
Add the new email block via the block library.
Feel free to add any other blocks just for fun.
Preview the project.  Enter something that's a mailformed address (the validator expects an @ and a valid . domain name) and click away from the email field.  A red error message will appear and the form will not submit.
Update the field with a vaild email address, see the error go away and submit will now work.
